pkgs/modules/rust/default.nix

```diff
@@ -1,60 +1,42 @@
-{ pkgs-unstable, lib, ... }:
+{ pkgs, lib, ... }:
 let
-  pkgs = pkgs-unstable;
-  cargoRun = pkgs.writeScriptBin "cargo_run" ''
-    if [ ! -f "$HOME/$REPL_SLUG/Cargo.toml" ]; then
-      NAME=$(echo $REPL_SLUG | sed -r 's/([a-z0-9])([A-Z])/\1_\2/g'| tr '[:upper:]' '[:lower:]')
-      ${pkgs.cargo}/bin/cargo init --name=$NAME
-    fi
-
-    ${pkgs.cargo}/bin/cargo run
-  '';
-  rust-version = lib.versions.majorMinor pkgs.rustc.version;
+  inherit (pkgs.fenix.stable) toolchain;
 in
 {
-  id = "rust-${rust-version}";
+  id = "rust-stable";
   name = "Rust Tools";
 
-  replit.packages = with pkgs; [
-    cargo
-    clang
-    rustc
-  ];
+  replit.dev.packages = with toolchain; [
+    toolchain
 
-  replit.dev.packages = with pkgs; [
-    rustfmt
-    rust-analyzer
+    pkgs.clang
   ];
 
   # TODO: should compile a binary to use in deployment and not include the runtime
   replit.runners.cargo = {
     name = "cargo run";
     language = "rust";
 
-    start = "${cargoRun}/bin/cargo_run";
+    start = "${toolchain}/bin/cargo run";
     fileParam = false;
   };
 
   replit.dev.languageServers.rust-analyzer = {
     name = "rust-analyzer";
     language = "rust";
 
-    start = "${pkgs.rust-analyzer}/bin/rust-analyzer";
+    start = "${toolchain}/bin/rust-analyzer";
+
+    initializationOptions = {
+      cargo.sysroot = "${toolchain}";
+    };
   };
 
   replit.dev.formatters.cargo-fmt = {
     name = "cargo fmt";
     language = "rust";
 
-    start = "${pkgs.cargo}/bin/cargo fmt";
-    stdin = false;
-  };
-
-  replit.dev.formatters.rustfmt = {
-    name = "rustfmt";
-    language = "rust";
-
-    start = "${pkgs.rustfmt}/bin/rustfmt $file";
+    start = "${toolchain}/bin/cargo fmt";
```

Comment:
I don't think we should make this change, because right now I think we have the contract that our formatters are about formatting a single file. I think this formats the while project, which might be surprising and have an unexpected algorithmic complexity from the Workspace's perspective.